misc-scripts/xref_mapping/XrefParser/VGNCParser.pm

```diff
@@ -21,93 +21,109 @@ package XrefParser::VGNCParser;
 
 use strict;
 use warnings;
-use File::Basename;
 use Carp;
-use base qw( XrefParser::HGNCParser);
+use Text::CSV;
 
-sub run {
+use parent qw( XrefParser::HGNCParser );
 
+sub run {
   my ($self, $ref_arg) = @_;
+
   my $source_id    = $ref_arg->{source_id};
   my $species_id   = $ref_arg->{species_id};
   my $files        = $ref_arg->{files};
-  my $verbose      = $ref_arg->{verbose};
-  my $dbi          = $ref_arg->{dbi};
-  $dbi = $self->dbi unless defined $dbi;
+  my $verbose      = $ref_arg->{verbose} // 0;
+  my $dbi          = $ref_arg->{dbi} // $self->dbi;
+
 
-  if((!defined $source_id) or (!defined $species_id) or (!defined $files) ){
-    croak "Need to pass source_id, species_id, files and rel_file as pairs";
+  if ( (!defined $source_id) or (!defined $species_id) or (!defined $files) ) {
+    croak 'Need to pass source_id, species_id and files';
   }
-  $verbose |=0;
 
-  my $file = @{$files}[0];
+  my $file = shift @{$files};
 
   my $count = 0;
 
   my $file_io = $self->get_filehandle($file);
 
   if ( !defined $file_io ) {
-    print "ERROR: Can't open VGNC file $file\n";
-    return 1;
+    croak "ERROR: Can't open VGNC file $file\n";
   }
 
   my $source_name = $self->get_source_name_for_source_id($source_id, $dbi);
+
   # Create a hash of all valid taxon_ids for this species
   my %species2tax = $self->species_id2taxonomy($dbi);
   my @tax_ids = @{$species2tax{$species_id}};
   my %taxonomy2species_id = map{ $_=>$species_id } @tax_ids;
 
-  # Skip header
-  $file_io->getline();
-
-  while ( $_ = $file_io->getline() ) {
-    chomp;
-    my @array = split /\t/x, $_;
-
-    my $taxon_id         = $array[0];
-    my $acc              = $array[1];
-    my $symbol           = $array[2];
-    my $name             = $array[3];
-    my $id               = $array[20];
-    my $previous_symbols = $array[9];
-    my $synonyms         = $array[11];
-
-    $previous_symbols =~ s/"//g;
-    $synonyms =~ s/"//g;
-
-    unless (exists ($taxonomy2species_id{$taxon_id})) { next; }
-
-    if ($id){              # Ensembl direct xref
-      $self->add_to_direct_xrefs({ stable_id  => $id,
-				   type       => 'gene',
-				   acc        => $acc,
-				   label      => $symbol,
-				   desc       => $name,
-                                   dbi        => $dbi,
-				   source_id  => $source_id,
-				   species_id => $species_id} );
-
-      $self->add_synonyms_for_hgnc( {source_id  => $source_id,
-                                     name       => $acc,
-                                     species_id => $species_id,
-                                     dbi        => $dbi,
-                                     dead       => $previous_symbols,
-                                     alias      => $synonyms});
+  my $input_file = Text::CSV->new({
+    sep_char       => "\t",
```

Comment:
Can you turn strict on here?

Reply:
no I can't...
last "column" in the source file is uniprot_ids, and that can be any number of columns (usually just one) containing a uniprot_id each.
The parser ignores the uniprot ids so it doesn't really matter (all previous fields are consistent), but turning strict on will complain about inconsistent number of fields.

Reply:
Eww